packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -63,15 +62,15 @@ pub trait BatchPool<B: Batch>: Send + Sync {
     /// Removes the [`Batch`] at position `index` within the pool, shifting all elements after it to
     /// the left.
     ///
-    /// Returns the batch id of the removed [`Batch`].
+    /// Returns the removed [`Batch`].
     ///
     /// # Panics
     ///
     /// - If `index` is out of bounds, or
     /// - if the `Batch` is currently borrowed as a [`BatchReadProxy`] or [`BatchWriteProxy`].
-    fn remove(&mut self, index: usize) -> String;
+    fn remove(&mut self, index: usize) -> RwLock<B>;
```

Comment:
As this needs to be unique at this point, can't we just return `B`?

Reply:
I think we thought that it was cheaper to pass the RwLock due to a wrong assumption that it was a pointer-type and wouldn't involve moving the Batch data itself. I agree, we should return B here.